js/ui/scheduler/appointmentPopup/popup.js

```diff
@@ -176,6 +176,14 @@ export class AppointmentPopup {
         );
     }
 
+    _setTimeZonesVisibility() {
+        if(this.scheduler.getEditingConfig().allowTimeZoneEditing) {
+            const isVisible = !this.state.appointment.data.allDay;
+            this.form.form.itemOption('undefined.startDateTimeZone', 'visible', isVisible);
+            this.form.form.itemOption('undefined.endDateTimeZone', 'visible', isVisible);
```

Comment:
It's better to use this.form.[dxForm](https://github.com/DevExpress/DevExtreme/blob/22_2/js/ui/scheduler/appointmentPopup/form.js#L73) instead of this.form.**form**